aplib.net-demo/Assets/Scripts/Entities/Player/PlayerLogic.cs

```diff
@@ -27,15 +31,20 @@ private void OnHurt(HealthComponent healthComponent, int amount)
         Debug.Log("Player took damage: " + amount);
     }
 
+    private void OnSelfRespawn (RespawnableComponent respawnableComponent)
+    {
+        Debug.Log("Player respawned!");
+        _healthComponent.Reset();
+    }
+
     /// <summary>
     /// On death, reset the rigidbody and health (respawn).
     /// </summary>
     /// <param name="healthComponent">The health component of the player.</param>.
     private void OnDeath(HealthComponent healthComponent)
     {
         Debug.Log("Player died!");
```

Comment:
Hoort niet bij jou code maar dit is een beetje een overbodige Debug.Log
```suggestion
    {
```